optuna/storages/_rdb/storage.py

```diff
@@ -222,15 +223,22 @@ def set_study_direction(self, study_id: int, direction: StudyDirection) -> None:
 
         with _create_scoped_session(self.scoped_session) as session:
             study = models.StudyModel.find_or_raise_by_id(study_id, session)
-
-            if study.direction != StudyDirection.NOT_SET and study.direction != direction:
+            current_direction = tuple(
```

Comment:
Just curious but is the cast to tuple necessary?

Reply:
When I worked on the MO integration, I assumed that the input was a tuple of `StudyDirection`, so I also cast `current_direction` to a tuple. However, it is not appropriate to make implicit assumptions about the input, so I will cast the input when MO integration. I will simply leave it as a list here.